engineio/src/asynchronous/async_transports/websocket_general.rs

```diff
@@ -86,30 +86,25 @@ impl Stream for AsyncWebsocketGeneralTransport {
     ) -> std::task::Poll<Option<Self::Item>> {
         let mut lock = ready!(Box::pin(self.receiver.lock()).poll_unpin(cx));
 
-        // poll on the underlying transport and map the result to the required type
-        lock.poll_next_unpin(cx)
-            .map(|option| {
-                option.map(|result| {
-                    result.map(|msg| {
-                        if msg.is_binary() {
-                            let data = msg.into_data();
-                            let mut msg = BytesMut::with_capacity(data.len() + 1);
-                            msg.put_u8(PacketId::Message as u8);
-                            msg.put(data.as_ref());
-
-                            msg.freeze()
-                        } else if msg.is_text() {
-                            Bytes::from(msg.into_data())
-                        } else {
-                            // in the edge case of a packet with a type other than text
-                            // or binary map it to an empty packet.
-                            // this always needs to get filtered out
-                            // by the user of the stream
-                            Bytes::new()
-                        }
-                    })
-                })
-            })
-            .map_err(Error::WebsocketError)
+        loop {
```

Comment:
suggestion: why loop? this match always returns.

Reply:
I refactored the loop in the newest commit. Generally the idea is to poll for a new packet (skip the current one) in case the packet has a type other than message or binary.

Reply:
If the type is other than text/binary wouldn't that be a Poll::Pending? See https://github.com/1c3t3a/rust-socketio/pull/168/files/31d3712f2021f738dc9daa386ddd1b50be8d2330..7dd1cd1d0b55094e71b058951e1e7a8e4f78fdba#r837707732